### PR TITLE
[TOOLS-1153] Team membership roles and team visibility

### PR DESCRIPTION
**Pull request by robersongomes**

### CODE REVIEW

**Comment by robersongomes** [comment]
whoever is going to review this PR, I suggest looking the commits one by one, cause it's a lot of code, and involves some refactorings and extractions

**Comment by robersongomes** [comment]
The first commit (TOOLS-1882) is the one I reverted on https://github.com/objective-solutions/taskboard/pull/466 so it doesn't need to be reviewed, as I just re-introduced it and fixed the cyclic dependency.

**Comment by robersongomes** [comment]
@taksan are you going to review this one?

**Comment by taksan** [comment]
> @taksan are you going to review this one?

I will. My first observation is why package-lock.json changed? I think there's something wrong with that

**Comment by robersongomes** [comment]
> I will. My first observation is why package-lock.json changed? I think there's something wrong with that

@taksan I'm not sure why Jhonny commited those files, but if you are on master and build the project, both package-lock.json get changed.

If you prefer, I can remove it from the commit they were changed.

File: src/test/java/objective/taskboard/team/TeamsControllerTest.java [hunk 1]
```diff
@@ -40,9 +43,9 @@ public void getTeams_ifLoggedUserIsntAdmin_returnNotFound() {
     }
 
     @Test
-    public void getTeams_ifLoggedUserIsAdmin_returnOkWithSortedValues() {
-        TeamsController subject = TeamsControllerMockBuilder.init()
-            .withTaskboardAdministrationPermission(true)
+    public void getTeams_ifLoggedUserHasTaskboardAdministrationPermission_returnOkWithSortedValues() {
+        TeamsController subject = teamsControllerBuilder()
+            .withTeamsEditViewPermission()
```
Comment (taksan) [inline src/test/java/objective/taskboard/team/TeamsControllerTest.java]: I'd prefer something like:
```
            loggedUser().
                hasTeamsEditViewPermission().and().
               .canAdminTeams()
                   .team("TASKBOARD")
                   .team("SDLC")
```

**Pull request merged**